### ISSUE TITLE
Gradient fills not properly rendered

### ISSUE THREAD

**Issue by ouening** [textext/textext#148]
**- Problem Description **
When I use textext to draw some pictures using pgfplots package, some error happens. Detailed description is listed bellow:
1. Firstly I modified the file `default_packages.tex` to support pgfplots:
```
\usepackage{amsmath,amsthm,amssymb,amsfonts}
\usepackage{color}
\usepackage{times}
\usepackage{epsfig}
\usepackage{graphicx}
\usepackage{amsmath}
\usepackage[lined,boxed,commentsnumbered, ruled]{algorithm2e}
\usepackage{bm}
\usepackage{multirow}
\usepackage{pgfplots}
\pgfplotsset{compat=1.16}
\usepackage{pgfplotstable}
\usepackage{verbatim}
\usepgfplotslibrary{patchplots}
\usepackage{tikz}
\usetikzlibrary{mindmap,shadows}
\usepackage[hidelinks,pdfencoding=auto]{hyperref}
\usepackage{smartdiagram}
\usepackage{fullpage}
\usepackage{fourier}
\usepgfplotslibrary{colorbrewer}
\usepackage{sansmath}
\usetikzlibrary{shadings,intersections}
```
2. Problem 1: No effect using **colorbar**, the code is:
```
\begin{tikzpicture}
\begin{axis}[
title={$x \exp(-x^2-y^2)$},
xlabel=$x$, ylabel=$y$,
colorbar horizontal,
small,
]
\addplot3 [
surf,
domain=-2:2,
domain y=-1.3:1.3,
] {exp(-x^2-y^2)*x};
\end{axis}
\end{tikzpicture}
```
the result is:
![no-colorbar](https://user-images.githubusercontent.com/15122811/67614995-fa039680-f7f8-11e9-9d9e-f757890f26a8.png)
As we can see that the colorbar has nothing, but no any warning or error happens.

3. Problem 2: Can't launch 3rd party executables (using contour gnuplot), I have installed gnuplot and add it to system paths, the code I used is:
```
\begin{tikzpicture}
\begin{axis}[
title={$x \exp(-x^2-y^2)$},
enlarge x limits,
view={0}{90},
xlabel=$x$, ylabel=$y$,
small,
]
\addplot3[
domain=-2:2,
domain y=-1.3:1.3,
contour gnuplot={number=14},
thick,
] {exp(-x^2-y^2)*x};
\end{axis}
\end{tikzpicture}
```
Error happens:
```
! Package pgfplots Error: sorry, plot file{tmp_contourtmp0.table} could not be 
opened.

See the pgfplots package documentation for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.42 ] {exp(-x^2-y^2)*x};
                         
No pages of output.
Transcript written on tmp.log.
```
According to the manual of pgfplots:
<img width="661" alt="pgfplots-contour-gnuplot" src="https://user-images.githubusercontent.com/15122811/67615059-e86ebe80-f7f9-11e9-8f13-3402e4aef98c.png">
I compiled it succesufully with `-shell-escape` argument in a tex file.

**- Hope for Solution**
1. Hope for support for additional argument for compiling
2. Can someone fix the problem 1?

**Comment by sizmailov** [comment]
@ouening, Thanks for report! Please report distinct bugs as separate issues next time. 

I was able to reproduce both bugs in TexText 0.11.0 & inkscape 0.92.4 (didn't check with inkscape 1.0alpha + dev version of TexText)

The first problem caused by some error on svg import generated by `pdf2svg`. I'll try to find out the root cause later. 

To fix second problem you need to change couple of lines in `~/.config/extensions/textext/__init__.py`: 

1. line 562: add `-shell-escape` option to *latex options 
```python
        LATEX_OPTIONS = ['-interaction=nonstopmode',
                         '-shell-escape',
                         '-halt-on-error']
```

2. line 617: reorder arguments order from 
from 
```python 
exec_command([tex_command, self.tmp('tex')] + self.LATEX_OPTIONS)
```
to
```python 
exec_command([tex_command] + self.LATEX_OPTIONS + [self.tmp('tex')])
```

@jcwinkler We may want to have configurable command line options, not necessary in GUI, just in `config.json`. 

All fixes should go to the next 0.12 release

**Comment by ouening** [comment]
> @ouening, Thanks for report! Please report distinct bugs as separate issues next time.
> 
> I was able to reproduce both bugs in TexText 0.11.0 & inkscape 0.92.4 (didn't check with inkscape 1.0alpha + dev version of TexText)
> 
> The first problem caused by some error on svg import generated by `pdf2svg`. I'll try to find out the root cause later.
> 
> To fix second problem you need to change couple of lines in `~/.config/extensions/textext/__init__.py`:
> 
> 1. line 562: add `-shell-escape` option to *latex options
> 
> ```python
>         LATEX_OPTIONS = ['-interaction=nonstopmode',
>                          '-shell-escape',
>                          '-halt-on-error']
> ```
> 
> 1. line 617: reorder arguments order from
>    from
> 
> ```python
> exec_command([tex_command, self.tmp('tex')] + self.LATEX_OPTIONS)
> ```
> 
> to
> 
> ```python
> exec_command([tex_command] + self.LATEX_OPTIONS + [self.tmp('tex')])
> ```
> 
> @jcwinkler We may want to have configurable command line options, not necessary in GUI, just in `config.json`.
> 
> All fixes should go to the next 0.12 release

Thank you very much!

**Comment by ouening** [comment]
> Found the reason for missing gradient issue, I'll submit PR for dev branch, so fix will be available for inkscape1.0 users

Could you tell me how to fix it? Thank u!

**Comment by sizmailov** [comment]
@ouening Please take a look at PR draft #149. Please note it's dev version and requires inkscape 1.0+.

**Comment by jcwinkler** [comment]
There seems to be a problem in Inkscape which prevents solving this issue completely under Windows, see https://gitlab.com/inkscape/inbox/issues/1260

**Comment by ouening** [comment]
> @ouening, Thanks for report! Please report distinct bugs as separate issues next time.
> 
> I was able to reproduce both bugs in TexText 0.11.0 & inkscape 0.92.4 (didn't check with inkscape 1.0alpha + dev version of TexText)
> 
> The first problem caused by some error on svg import generated by `pdf2svg`. I'll try to find out the root cause later.
> 
> To fix second problem you need to change couple of lines in `~/.config/extensions/textext/__init__.py`:
> 
> 1. line 562: add `-shell-escape` option to *latex options
> 
> ```python
>         LATEX_OPTIONS = ['-interaction=nonstopmode',
>                          '-shell-escape',
>                          '-halt-on-error']
> ```
> 
> 1. line 617: reorder arguments order from
>    from
> 
> ```python
> exec_command([tex_command, self.tmp('tex')] + self.LATEX_OPTIONS)
> ```
> 
> to
> 
> ```python
> exec_command([tex_command] + self.LATEX_OPTIONS + [self.tmp('tex')])
> ```
> 
> @jcwinkler We may want to have configurable command line options, not necessary in GUI, just in `config.json`.
> 
> All fixes should go to the next 0.12 release

In Inscape1.0 using TexText1.0.1, we should modify file `textext/base.py` in **line 485 and line 527**. I don't know how to configure `config.josn` instead of hacking the code.
